test/jdk/java/awt/Checkbox/CheckboxCheckerScalingTest.java

```diff
@@ -72,7 +74,8 @@ public static void main(String[] args) throws Exception {
                 check: {
                     for (int i = 0; i < imageAfterChecked.getHeight(); i++) {
                         for (int j = 0; j < imageAfterChecked.getWidth(); j++) {
-                            if (Color.black.getRGB() == imageAfterChecked.getRGB(i, j)) {
+                            Color pixelColor = new Color(imageAfterChecked.getRGB(i, j));
+                            if (compareColor(pixelColor)) {
```

Comment:
Current pixel comparison suggests that the test passes even if one pixel satisfies the condition. This may give some false positive result.
Is it possible to test all pixel and then check the number of pixels passed against total number of pixels compared and if it is more than some percentage value ( may be ~85-90) then return true otherwise false.

This may lead to avoid the tolerance check if sufficient number of pixels are exactly black in color.

Reply:
Checker mark will be less than 50% of the image. The image captured is around 8x8 pixels, in that we hardly can find 10% of black color (though with tolerance of 5-10 value). Since we are making decisions based on black color which is the actual checker color with tolerance, checking for one such match would be fine.